### ISSUE TITLE
OPM world domination is not happening and luarocks is deprecated

### ISSUE THREAD

**Issue by rulatir** [openresty/openresty#471]
At the time of this writing, maybe 20% of CoolStuffThatEveryoneWants™ is available in OPM. Everything else still requires luarocks to install.

Please undeprecate luarocks and make OPM play nicely like a good newcomer should until it is ready for world domination.

**Comment by agentzh** [comment]
Yeah, we'll improve OPM and there are a lot of features planned. We don't want to officially support LuaRocks at least not in its original form. The user can still use LuaRocks, it's just not officially supported.

**Comment by zcaudate** [comment]
@agentzh: Is there currently a way to install luarocks modules using opm?

**Comment by zcaudate** [comment]
I want to use `resty` purely for its runtime for shell scripting (using the `ngx.threads`). I'm on osx and removed the `brew` installations of `lua`/`luarocks`/`luajit` because of all the difficulties with incompatible lua versions. 

For example, `luarocks` has a dependency on` lua5.4` and I needed to try out the `cjson` for `lua5.1`. Being new to lua, It literally took me 2 hours of trial and error to figure out that it was easier to copy and paste the source code of an earlier cjson version into my directory.

I'd actually prefer to just install the `openresty` package via `brew` and go with the packaged tools via `opm`. I had a google giggle at the title of the issue. I don't think it's really about world domination but just making the package managers compatible and easy to use.

**Comment by qhaas** [comment]
The main 'third-party' module my projects use is [`lua-resty-openidc`](https://github.com/zmartzone/lua-resty-openidc/blob/v1.8.0/README.md?plain=1#L59), whose documentation states `Older versions of lua-resty-openidc could also be installed using opm but this is no longer supported`.  That is at odds with the [OpenResty statement](https://openresty.org/en/using-luarocks.html) `Use of LuaRocks with OpenResty is strongly discouraged`.

If one tries to install the dependencies of `lua-resty-openidc` using OPM, luarocks doesn't seem to recognize them as being already installed, which isn't surprising given the mixing and matching of package managers.

The above suggestion of allowing OPM to install packages from luarocks repository, and deal with dependency issues, would be a logical path forward.
